rero_ils/utils.py

```diff
@@ -80,3 +81,63 @@ def get_agg_config(index_name, field):
             current_app.config.get('RERO_ILS_DEFAULT_AGGREGATION_SIZE')
         )
     ))
+
+
+def git_get_current_branch():
+    """Get current GIT branch or None.
+
+    Use only files to determine in which branch we are.
+
+    :return: string of current GIT branch or None
+    """
+    res = None
+    current_dir = os.path.dirname(__file__)
+    filepath = os.path.join(current_dir, '../.git/HEAD')
+    if os.path.exists(filepath):
+        with open(filepath, 'r') as headfile:
+            res = headfile.read().strip().split('/')[-1]
+    return res
+
+
+def git_get_last_commit_hash():
+    """Get last GIT commit hash.
+
+    Use only files to determine what's the last commit hash.
+
+    :return: string of last GIT commit hash or None
+    """
+    res = None
+    current_branch = git_get_current_branch()
+    if current_branch:
+        current_dir = os.path.dirname(__file__)
+        filepath = os.path.join(
+            current_dir,
+            '../.git/refs/heads/', current_branch)
+        if os.path.exists(filepath):
+            with open(filepath, 'r') as branchfile:
+                res = branchfile.read().strip()
+    return res
+
+
+def git_short_hash():
+    """Get current project git commit hash. Only 8 first chars.
+
+    1. use GIT_HASH environment variable.
+    2. If no GIT_HASH environment variable, use .git/refs/heads/ directory
+        - master if production
+        - dev if development
+        - only 8 first chars
+    3. If no file: return empty string.
+
+    :return: string of last GIT commit hash, 8 first chars. Or None.
+    """
+    res = ''
+    # Attempt 1
+    _git_hash = os.environ.get("GIT_HASH", default=None)
```

Comment:
I propose to not use environment variable but flask config instead: `RERO_ILS_APP_GIT_HASH` or `RERO_ILS_APP_GIT_VERSION`. This will avoid variable injection in the template file as config is available in templates. It is also more flexible as we can use env variable or config file.